### ISSUE TITLE
Compile error with cudnn 7.0

### ISSUE THREAD

**Issue by dbermond** [facebookarchive/caffe2#1025]
I'm getting the following error when trying to compile caffe2 with cudnn 7.0:
```
[ 83%] Building CXX object caffe2/CMakeFiles/Caffe2_GPU.dir/operators/recurrent_op_cudnn.cc.o
In file included from /storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/operators/recurrent_op_cudnn.h:4:0,
                 from /storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/operators/recurrent_op_cudnn.cc:1:
/storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/operators/recurrent_op_cudnn.cc: In instantiation of ‘void caffe2::RecurrentBaseOp<T>::initialize(const caffe2::Tensor<caffe2::CUDAContext>&, caffe2::Tensor<caffe2::CUDAContext>*, caffe2::Tensor<caffe2::CUDAContext>*, caffe2::Tensor<caffe2::CUDAContext>*, caffe2::Tensor<caffe2::CUDAContext>*) [with T = float]’:
/storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/operators/recurrent_op_cudnn.cc:388:13:   required from ‘bool caffe2::RecurrentParamAccessOp<T, mode>::RunOnDevice() [with T = float; caffe2::RecurrentParamOpMode mode = (caffe2::RecurrentParamOpMode)1u]’
/storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/operators/recurrent_op_cudnn.cc:595:1:   required from here
/storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/operators/recurrent_op_cudnn.cc:118:40: error: cannot convert ‘cudnnRNNDescriptor_t {aka cudnnRNNStruct*}’ to ‘cudnnHandle_t {aka cudnnContext*}’ for argument ‘1’ to ‘cudnnStatus_t cudnnSetRNNDescriptor(cudnnHandle_t, cudnnRNNDescriptor_t, int, int, cudnnDropoutDescriptor_t, cudnnRNNInputMode_t, cudnnDirectionMode_t, cudnnRNNMode_t, cudnnRNNAlgo_t, cudnnDataType_t)’
     CUDNN_ENFORCE(cudnnSetRNNDescriptor(
                                        ^
/storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/core/common_cudnn.h:64:28: note: in definition of macro ‘CUDNN_ENFORCE’
     cudnnStatus_t status = condition;                     \
                            ^
In file included from /storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/core/common_cudnn.h:7:0,
                 from /storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/operators/recurrent_op_cudnn.h:4,
                 from /storage/linux/abs/caffe2-git/src/caffe2-git/caffe2/operators/recurrent_op_cudnn.cc:1:
/opt/cuda/include/cudnn.h:1553:8: note: class type ‘cudnnRNNStruct’ is incomplete
 struct cudnnRNNStruct;
        ^
[ 84%] Building CXX object caffe2/CMakeFiles/Caffe2_GPU.dir/operators/relu_op_cudnn.cc.o
[ 84%] Building CXX object caffe2/CMakeFiles/Caffe2_GPU.dir/operators/softmax_op_cudnn.cc.o
make[2]: *** [caffe2/CMakeFiles/Caffe2_GPU.dir/build.make:1012: caffe2/CMakeFiles/Caffe2_GPU.dir/operators/recurrent_op_cudnn.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:439: caffe2/CMakeFiles/Caffe2_GPU.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

Compiling with cudnn 6.0.21 works fine.

**System information:**
**OS:** Arch Linux x86_64
**caffe2:** git master (currently at 0.8.0.r126.g21d0d767)
**compiler:** gcc 5.4.0
**cudnn:** 7.0.1
**cuda:** 8.0.61

**Comment by slayton58** [comment]
/cc @Yangqing as he's issued a PR that may well fix this.

cuDNN v7 support has not yet been added fully - I'll be issuing PRs in the next day or so

**Comment by Yangqing** [comment]
Yep, this is fixed in 707aed36e89ab9e2041de25166a4930fc4e24ee7 .
